src/main/java/io/cdap/plugin/dlp/SensitiveRecordRedaction.java

```diff
@@ -301,6 +315,19 @@ private DlpServiceSettings getSettings() throws IOException {
       return transformationConfigs;
     }
 
+    public HashSet<String> getRequiredFields() {
+
+      HashSet requiredFields = new HashSet<String>();
+      try {
+        List<DlpFieldTransformationConfig> transformationConfigs = parseTransformations();
+        for (DlpFieldTransformationConfig config : transformationConfigs) {
+          requiredFields.addAll(config.getRequiredFields());
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
```

Comment:
Should be logged and propogated ? e.printStackTrace seems wrong no ?

Reply:
Don't use `printStackTrace`, always use a Logger.

Reply:
Good catch, updated it to log the error. No need to propagate since this is an optimization so if this fails then we can default to the regular way of doing things by returning null.